### ISSUE TITLE
Delimiter for paragraph

### ISSUE THREAD

**Issue by Ducasse** [pillar-markup/Microdown#298]
Right now a paragraph is ended by a newline and Pavel was  telling me that it would be better to have cr. 

I do not know but I took note about it. 
We can discuss this point.

**Comment by Ducasse** [comment]
In macDown this is the same as ours. 

in GH

this is a new pargarpha wiht super mogjk kjh jkh jkh kjh kjh kjhkjh kjh kjh kjh kjhkjh kjh kjh kjh kjh kjhkj hk kjh
jhhjhjhkjhkjhkjh
jkhkjhjkhkjhjkhjk

khjkjhjkkjhjkjkhj

**Comment by kasperosterbye** [comment]
Most paragraphs are so short I do not know if list or string is better - I am fine with both. You need to concatenate the lines into a list before doing the inline parsing as e.g. bold can stretch over multiple lines
this is **bold
and so is this** is what I mean

I do not understand the comment about newLine as paragraphs are ended by String cr as far as I can see. Is he refering to the Text generated by MicRichComposer? Or is it the latex generator? Or something else?

**Comment by Ducasse** [comment]
Good remark for bold.
Now it means that 
```
this is **bold
and so is this** is what I mean
```
should be represented as 

```
this is **bold and so is this** is what I mean
```
because you want to render paragraphs as a nice block

```
jkhkjh kjh jkh kjh kjh kjh kjh kjh kjh kjh kjh kjh kjh iuy iuy
 uyf y jk hj lkklj ;kl  kl;k  kl; ;lkkjgjhg g f hf
 hfhf hgf hgfhgfhgfhgfhgfhgf
 hgfhg fhgf hgfhgf 
```
should be renderer as 

jkhkjh kjh jkh kjh kjh kjh kjh kjh kjh kjh kjh kjh kjh iuy iuy uyf y jk hj lkklj ;kl  kl;k  kl; ;lkkjgjhg g f hf
 hfhf hgf hgfhgfhgfhgfhgfhgf hgfhg fhgf hgfhgf 

and not four lines stopping before the end. 

So now I understand what we should do.
We keep the delimiter of paragraph as it is: an empty line like GH and others MD.  

Now the paragraph text should not be managed as a concatenation of line + one crs. 
Because this is the job of the renderer to decide and per default the renderer should take the line and produce a paragraph without cr in the middle. 

In addition I would like to have more newlines emitted in the micRichTextComposer to clearly see the paragraph.

**Comment by Ducasse** [comment]
Is it clearer?

**Comment by kasperosterbye** [comment]
> On 26 Nov 2021, at 18.36, StéphaneDucasse ***@***.***> wrote:
> 
> In addition I would like to have more newlines emitted in the micRichTextComposer to clearly see the paragraph.

Emission of newlines will be controlled by the textStyle object in the next version PR you will see. I imagine there are times when compact is nice, some times where an extra newline is nice, and sometime where you actually want newline and indent (like in some Latex styles).

The canvas operations for newline and newlineIfNotAlready is using the textStyler.

Best,

Kasper

**Comment by Ducasse** [comment]
Ok have a look at the PR I did.

**Comment by kasperosterbye** [comment]
The option to change how lines are merged have now been moved to the text styler, and merged in PR #307
